src/pytest_replay/__init__.py

```diff
@@ -84,15 +109,27 @@ def pytest_collection_modifyitems(self, items, config):
             config.hook.pytest_deselected(items=deselected)
             items[:] = remaining
 
-    def append_test_to_script(self, nodeid):
+    def append_test_to_script(self, nodeid, line):
         suffix = "-" + self.xdist_worker_name if self.xdist_worker_name else ""
         fn = os.path.join(self.dir, self.base_script_name + suffix + self.ext)
         with open(fn, "a", encoding="UTF-8") as f:
-            f.write(nodeid + "\n")
+            f.write(line + "\n")
+            f.flush()
             self.written_nodeids.add(nodeid)
 
 
+class DeferPlugin:
+    def pytest_configure_node(self, node):
+        node.workerinput["replay_start_time"] = node.config.replay_start_time
+
+
 def pytest_configure(config):
+    if hasattr(config, "workerinput"):
```

Comment:
All of this should only be done if `ReplayPlugin` is actually registered, otherwise it won't be used at all. 😉